### ISSUE TITLE
Problem with interpolations in delta extraction

### ISSUE THREAD

**Issue by andreicuceu** [igmhub/picca#1007]
While investigating a different problem with the true continuum code (probably coming in a separate issue soon), I plotted the weights of some forests. These are coming from an uncontaminated DESI Y5 mock. I attached a figure with a few forests I've picked based on their SNR. The problem is that the high SNR forests appear to have a step function in their weights. I then overplotted 1/var_lss, from which you can see that the steps align with the bins in var_lss.

This is caused by the interpolation of var_lss which is done with interp1d using the "nearest" option instead of linear or cubic as I would have expected. So a straight forward way to address this would be to change all interpolations to cubic. This would definitely break all the tests, so I wanted to get everyone's opinion before opening a pull request.

![forest_weights](https://github.com/igmhub/picca/assets/28951059/fef18bbc-06f0-48c8-b301-56b9060ba33f)

**Comment by iprafols** [comment]
Hi Andrei, 

I agree we should change this. I was planning to implement our own interpolator following @p-slash  suggestion to speed up the searches, and I can implement a cubic interpolation too. That said, I would go ahead with the changes first, as these seem simple enough and preparing a new interpolator might take a bit of time (specially since I'm busy elsewhere too)

**Comment by iprafols** [comment]
Also, this seems an important fix that would break backward compatibility. Thus, I would create a new tag once this is merged.

**Comment by andreicuceu** [comment]
Okey, sounds good to me. I'll make a pull request.

**Comment by andreufont** [comment]
Can we close this issue, @andreicuceu ?

**Comment by andreicuceu** [comment]
Yes, happy to close as this was fixed by #1008
